src/components/application_manager/test/commands/mobile/create_interaction_choice_set_test.cc

```diff
@@ -0,0 +1,612 @@
+
+  command_->on_event(event);
+}
+
+TEST_F(CreateInteractionChoiceSetRequestTest,
+       OnEvent_ValidVrNoErrorAndExpectedChoiceLessThanReceiveChoice_SUCCESS) {
+  Event event(hmi_apis::FunctionID::VR_AddCommand);
+
+  (*message_)[am::strings::params][am::strings::correlation_id] =
+      kCorrelationId;
+  (*message_)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::eType::WARNINGS;
+
+  FillMessageFieldsItem1(message_);
+
+  (*message_)[am::strings::msg_params][am::strings::interaction_choice_set_id] =
+      kChoiceSetId;
+  ON_CALL(mock_app_manager_, application(_)).WillByDefault(Return(app_));
+
+  EXPECT_CALL(*message_helper_mock_, VerifyImage(_, _, _))
+      .WillRepeatedly(Return(mobile_apis::Result::SUCCESS));
+
+  smart_objects::SmartObject* choice_set_id = NULL;
+  EXPECT_CALL(*app_, FindChoiceSet(kChoiceSetId))
+      .WillOnce(Return(choice_set_id));
+
+  EXPECT_CALL(mock_app_manager_, GenerateGrammarID())
+      .WillOnce(Return(kGrammarId));
+  EXPECT_CALL(*app_, AddChoiceSet(kChoiceSetId, _));
+  ON_CALL(mock_app_manager_, GetNextHMICorrelationID())
+      .WillByDefault(Return(kCorrelationId));
+  command_->Run();
+
+  FillMessageFieldsItem2(message_);
+
+  EXPECT_CALL(mock_app_manager_, updateRequestTimeout(_, _, _)).Times(0);
+  EXPECT_CALL(mock_app_manager_, TerminateRequest(_, _));
+  event.set_smart_object(*message_);
+  command_->on_event(event);
+}
+
+TEST_F(CreateInteractionChoiceSetRequestTest,
+       OnTimeOut_InvalidErrorFromHMI_UNSUCCESS) {
+  Event event(hmi_apis::FunctionID::VR_AddCommand);
+
+  (*message_)[am::strings::params][am::strings::correlation_id] =
+      kCorrelationId;
+  (*message_)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::eType::SUCCESS;
+
+  FillMessageFieldsItem1(message_);
+  (*message_)[am::strings::msg_params][am::strings::interaction_choice_set_id] =
+      kChoiceSetId;
+
+  ON_CALL(mock_app_manager_, application(_)).WillByDefault(Return(app_));
+
+  EXPECT_CALL(*message_helper_mock_, VerifyImage(_, _, _))
+      .WillRepeatedly(Return(mobile_apis::Result::SUCCESS));
+
+  smart_objects::SmartObject* choice_set_id = NULL;
+  EXPECT_CALL(*app_, FindChoiceSet(kChoiceSetId))
+      .WillOnce(Return(choice_set_id));
+
+  EXPECT_CALL(mock_app_manager_, GenerateGrammarID())
+      .WillOnce(Return(kGrammarId));
+  EXPECT_CALL(*app_, AddChoiceSet(kChoiceSetId, _));
+  ON_CALL(mock_app_manager_, GetNextHMICorrelationID())
+      .WillByDefault(Return(kCorrelationId));
+  command_->Run();
+
+  FillMessageFieldsItem2(message_);
+  EXPECT_CALL(mock_app_manager_, updateRequestTimeout(_, _, _)).Times(0);
+  EXPECT_CALL(mock_app_manager_, TerminateRequest(_, _));
+  event.set_smart_object(*message_);
+  command_->on_event(event);
+
+  EXPECT_CALL(mock_app_manager_, application(_)).WillOnce(Return(app_));
+  EXPECT_CALL(mock_app_manager_,
+              ManageMobileCommand(_, am::commands::Command::ORIGIN_SDL))
+      .Times(2);
```

Comment:
@LevchenkoS , this test has been simplified in e17c66c, and need to check only one call. Checked.